src/Nest/Modules/SnapshotAndRestore/Repositories/S3Repository.cs

```diff
@@ -7,14 +7,26 @@ public interface IS3Repository : IRepository<IS3RepositorySettings> { }
 
 	public class S3Repository : IS3Repository
 	{
-		public S3Repository(IS3RepositorySettings settings) => Settings = settings;
+		public S3Repository(S3RepositorySettings settings) => Settings = settings;
 
 		public IS3RepositorySettings Settings { get; set; }
 		public string Type { get; } = "s3";
 	}
 
 	public interface IS3RepositorySettings : IRepositorySettings
 	{
+		[Obsolete("Removed in Elasticsearch 6.0, please specify secure settings in the keystore. See https://www.elastic.co/guide/en/elasticsearch/reference/6.x/secure-settings.html")]
```

Comment:
slight nitpick: Not overly keen on links in the attributes as they can drift. Maybe just
```
Removed in Elasticsearch 6.0. Please specify secure settings in the Elasticsearch keystore.
```
?

Reply:
:+1: